spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/procedures/FastForwardBranchProcedure.java

```diff
@@ -77,12 +79,13 @@ public InternalRow[] call(InternalRow args) {
     return modifyIcebergTable(
         tableIdent,
         table -> {
-          long currentRef = table.currentSnapshot().snapshotId();
+          Snapshot currentSnapshot = table.snapshot(source);
+          Preconditions.checkArgument(
+              currentSnapshot != null, "Branch to fast forward does not exist: %s", source);
```

Comment:
minor: would it make sense to say `Branch to fast-forward...` here rather than `Branch to fast forward...`?